### PR TITLE
blockdev_mirror_qemuio_target:qemu-io target during mirror

### PR DESCRIPTION
**Pull request by aliang123**
qemu-io target image during mirror

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2121264

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_mirror_qemuio_target.q35: PASS (41.61 s)

**Comment by aliang123** [comment]
@zhencliu @MiriamDeng, can you help review this patch? thanks.

File: qemu/tests/blockdev_mirror_qemuio_target.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        image = self.disk_define_by_params(image_params, tag)
+        filename = image.image_filename
+        fmt = image.image_format
+        qemu_io = utils_misc.get_qemu_io_binary(self.params)
+        qemuio_cmd = self.params.get("qemuio_cmd") % (qemu_io, fmt, filename)
+        try:
+            process.run(qemuio_cmd, shell=True)
+        except process.CmdError as e:
+            if self.params["error_msg"] not in e.result.stderr.decode():
+                self.test.fail("Write to image that using by another process")
```
Comment (zhencliu) [inline qemu/tests/blockdev_mirror_qemuio_target.py:25]: Suggest to print the e.result.stderr here if we happen to hit an unknown error or if the err msg changes

File: qemu/tests/blockdev_mirror_qemuio_target.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        else:
+            self.test.fail("Can qemu-io a using image")
```
Comment (zhencliu) [inline qemu/tests/blockdev_mirror_qemuio_target.py:27]: Be careful here, if we can qemu-io a busy image(product bug), then qemu-io will never stop, so we never come here, is there a way to run qemu-io in certain time?  maybe
qemu-io -c quit  -f raw /tmp/aaa ?

File: qemu/tests/blockdev_mirror_qemuio_target.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    Qemuio target image during mirror
+
+    test steps:
+        1. boot VM with 2G data disk
+        2. format the data disk and mount it
+        3. create a file
+        4. add a local fs image for mirror to VM via qmp commands
+        5. do blockdev-mirror
+        6. qemu-io target image during mirror
```
Comment (zhencliu) [inline qemu/tests/blockdev_mirror_qemuio_target.py:44]: add comment that we must hit error after step6

File: qemu/tests/cfg/blockdev_mirror_qemuio_target.cfg [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    backup_options_data1 = sync
+    sync = full
+    storage_pools = default
+    storage_pool = default
+
+    image_size_data1 = 2G
+    image_size_mirror1 = ${image_size_data1}
+    #set target:raw here to cover raw->raw, qcow2->raw mirror,
+    #other two scenarios covered in other mirror cases
+    image_format_mirror1 = qcow2
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_mirror_qemuio_target.cfg:22]: looks target is a qcow2 image

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_mirror_qemuio_target.q35: PASS (45.85 s)

File: qemu/tests/cfg/blockdev_mirror_qemuio_target.cfg (hunk 4 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_mirror_qemuio_target.cfg]: Have updated, thanks @zhencliu

File: qemu/tests/blockdev_mirror_qemuio_target.py (hunk 3 above)
Comment (aliang123) [reply qemu/tests/blockdev_mirror_qemuio_target.py]: Have updated, thanks @zhencliu

File: qemu/tests/blockdev_mirror_qemuio_target.py (hunk 2 above)
Comment (aliang123) [reply qemu/tests/blockdev_mirror_qemuio_target.py:28]: Have updated, thanks @zhencliu

File: qemu/tests/blockdev_mirror_qemuio_target.py (hunk 1 above)
Comment (aliang123) [reply qemu/tests/blockdev_mirror_qemuio_target.py]: Have updated, thanks @zhencliu

**Comment by zhencliu** [comment]
LGTM

**Comment by MiriamDeng** [comment]
Will give results today. Thanks

**Pull request merged**